src/axom/primal/operators/detail/intersect_patch_impl.hpp

```diff
@@ -0,0 +1,248 @@
+ * \param [in] u_offset The offset in parameter space for \a patch in the u direction
+ * \param [in] u_scale The scale in parameter space for \a patch in the u direction
+ * \param [in] v_offset The offset in parameter space for \a patch in the v direction
+ * \param [in] v_scale The scale in parameter space for \a patch in the v direction
+ * \param [in] sq_tol Numerical tolerance for physical distances
+ * \param [in] EPS Numerical tolerance in parameter space
+ *
+ * A ray can only intersect a Bezier patch if it intersects its bounding box.
+ * The base case of the recursion is when we can approximate the patch as
+ * bilinear, where we directly find their intersections. Otherwise,
+ * check for intersections recursively after bisecting the patch in each direction.
+ *
+ * \note This detail function returns all found intersections within EPS of parameter space,
+ *  including identical intersections reported by each subdivision. 
+ * The calling `intersect` routine should remove duplicates and enforce half-open behavior. 
+ * 
+ * \note This function assumes that all intersections have multiplicity 
+ *  one, i.e. does not find tangencies. 
+ *
+ * \return True if the line intersects the patch, False otherwise
+ * \sa intersect_bezier
+ */
+template <typename T>
+bool intersect_line_patch(const Line<T, 3> &line,
+                          const BezierPatch<T, 3> &patch,
+                          axom::Array<T> &tp,
+                          axom::Array<T> &up,
+                          axom::Array<T> &vp,
+                          int order_u,
+                          int order_v,
+                          double u_offset,
+                          double u_scale,
+                          double v_offset,
+                          double v_scale,
+                          double sq_tol,
+                          double EPS,
+                          bool isRay);
+
+//------------------------------ IMPLEMENTATIONS ------------------------------
+
+template <typename T>
+bool intersect_line_patch(const Line<T, 3> &line,
+                          const BezierPatch<T, 3> &patch,
+                          axom::Array<T> &tp,
+                          axom::Array<T> &up,
+                          axom::Array<T> &vp,
+                          int order_u,
+                          int order_v,
+                          double u_offset,
+                          double u_scale,
+                          double v_offset,
+                          double v_scale,
+                          double sq_tol,
+                          double EPS,
+                          bool isRay)
+{
+  using BPatch = BezierPatch<T, 3>;
+
+  // Check bounding box to short-circuit the intersection
+  //  Need to expand the box a bit so that intersections near subdivision boundaries
+  //  are accurately recorded
+  Point<T, 3> ip;
+  if(!intersect(line, patch.boundingBox().scale(1.5), ip))
```

Comment:
In the general case, almost definitely, one just has to make sure that the scaling factor is reasonably consistent with the given tolerances. In retrospect, this might be easier to pull off using `BoundingBox::expand()` since the units there are the same as `sq_tol`. This might be even more relevant down the line when I need to do more extrapolation of surfaces to find near misses, but I can figure out the deeper details when it comes up.